marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync.hs

```diff
@@ -49,10 +49,11 @@ data ChainSyncDependencies = ChainSyncDependencies
 
 chainSync :: Component IO ChainSyncDependencies ()
 chainSync = proc ChainSyncDependencies{..} -> do
-  let DatabaseQueries{..} = databaseQueries
-  NodeClient{..} <- nodeClient -< NodeClientDependencies{..}
   let rateLimit = persistRateLimit
-  ChainStore{..} <- chainStore -< ChainStoreDependencies{..}
+  rec
+    let DatabaseQueries{..} = chainStoreDatabaseQueries
+    NodeClient{..} <- nodeClient -< NodeClientDependencies{..}
+    ChainStore{..} <- chainStore -< ChainStoreDependencies{..}
```

Comment:
This is calculating a fix point for some value, right? Presumably the fields in `NodeClient{..}` and `ChainStore{..}`.

Reply:
Yeah, Exactly - `NodeClientDependencies` depends on a value from `chainStoreDatabaseQueries`, and `ChainStoreDependencies` depends on a value from `NodeClient`